internal/provider/data_source_certificate.go

```diff
@@ -24,18 +25,26 @@ func dataSourceCertificate() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"url": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 				Description: "URL of the endpoint to get the certificates from. " +
 					fmt.Sprintf("Accepted schemes are: `%s`. ", strings.Join(SupportedURLSchemesStr(), "`, `")) +
 					"For scheme `https://` it will use the HTTP protocol and apply the `proxy` configuration " +
 					"of the provider, if set. For scheme `tls://` it will instead use a secure TCP socket.",
 				ValidateDiagFunc: validation.ToDiagFunc(validation.IsURLWithScheme(SupportedURLSchemesStr())),
+				ConflictsWith:    []string{"content"},
+			},
+			"content": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The content of the PEM encoded certificate",
```

Comment:
Like for other resources and data sources, please link in the descriptions the RFCs describing the format of PEM expected here ([examples](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key#private_key_pem)).

Also, please terminate the description with a `.`.

Reply:
Also, as `encoded` is a loaded term, please use the `PEM format` or `PEM formatted` expression here, in consistency with the other resources.